src/ResourceManager/Sql/Commands.Sql/Failover Group/Cmdlet/AzureSqlFailoverGroupCmdletBase.cs

```diff
@@ -43,5 +33,49 @@ protected override AzureSqlFailoverGroupAdapter InitModelAdapter(AzureSubscripti
         {
             return new AzureSqlFailoverGroupAdapter(DefaultProfile.Context);
         }
+
+        /// <summary>
+        /// Interprets and returns the lossy automatic failover grace period provided by the user.
+        /// </summary>
+        /// <returns>The grace period</returns>
+        protected int? ComputeEffectiveGracePeriod(FailoverPolicy policy, int? originalGracePeriod)
+        {
+            int? gracePeriod = null;
+
+            // Respect both parameters, but the non-obselete one takes precedence.
+            foreach (var parameterName in new[] { "GracePeriodWithDataLossHours", "GracePeriodWithDataLossHour" })
+            {
+                object parameterValue;
+                if (MyInvocation.BoundParameters.TryGetValue(parameterName, out parameterValue))
+                {
+                    gracePeriod = (int)parameterValue;
+                    break;
+                }
+            }
+
+            if (!gracePeriod.HasValue && policy == FailoverPolicy.Automatic)
+            {
+                // If the policy is Automatic, the grace period must be non-null. If this is an update and the grace
+                // period was non-null, use the existing grace period. Otherwise, use a default of 1.
+                gracePeriod = originalGracePeriod ?? 1;
```

Comment:
This is a subjective choice we made in favor of user convenience. What exactly do you mean by "safe?" Avoiding data loss unless the user very explicitly opts in? Also, this would arguably be a breaking change since previously the command would have worked without specifying the grace period.